#! /bin/sh -e
## 17_fix_qfontdatabase_cpp.dpatch by Lars Knoll <lars@trolltech.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix some stuff with regards to font handling

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -ruN qt-x11-free-3.3.4.old/src/kernel/qfontdatabase_x11.cpp qt-x11-free-3.3.4/src/kernel/qfontdatabase_x11.cpp
--- qt-x11-free-3.3.4.old/src/kernel/qfontdatabase_x11.cpp	2005-01-21 17:16:11.000000000 +0000
+++ qt-x11-free-3.3.4/src/kernel/qfontdatabase_x11.cpp	2005-03-18 15:16:58.544321304 +0000
@@ -961,8 +961,7 @@
         }
 #endif
 	QtFontEncoding *enc = size->encodingID( -1, 0, 0, 0, 0, TRUE );
-	enc->pitch = ( spacing_value >= XFT_CHARCELL ? 'c' :
-		       ( spacing_value >= XFT_MONO ? 'm' : 'p' ) );
+	enc->pitch = (f->fixed ? 'm' : 'p');
     }
 
     XftFontSetDestroy (fonts);
